src/dune/context.ml

```diff
@@ -151,7 +151,20 @@ let opam_config_var ~env ~cache var =
         | Error _ ->
             None ) )
 
-let which ~cache ~path x = Table.find_or_add cache x ~f:(Bin.which ~path)
+let best_prog dir prog =
+  let fn = Path.relative dir (prog ^ ".opt" ^ Bin.exe) in
+  if Bin.exists fn then
+    Some fn
+  else
+    let fn = Path.relative dir (prog ^ Bin.exe) in
```

Comment:
Option.some_if makes this a bit cleaner.

Reply:
Here I'm not convinced; if I use `Option.some_if` around the first call to `Bin.exists` then I need to pattern match on the result, which is not nice. And if I only use it around the second call to `Bin.exists` then the code loses its symmetry.

Reply:
The symmetry was lost on me :) anyway it's just a suggestion, feel free to leave as is.